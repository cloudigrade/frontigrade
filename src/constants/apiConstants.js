const API_ACCOUNT_ARN = 'account_arn';
const API_ACCOUNT_NAME = 'name';
const API_RESOURCE_TYPE = 'resourcetype';
const API_AWS_ACCOUNT_ID = 'aws_account_id';

const apiTypes = {
  API_ACCOUNT_ARN,
  API_ACCOUNT_NAME,
  API_RESOURCE_TYPE,
  API_AWS_ACCOUNT_ID
};

export { apiTypes as default, apiTypes, API_ACCOUNT_ARN, API_ACCOUNT_NAME, API_RESOURCE_TYPE, API_AWS_ACCOUNT_ID };
