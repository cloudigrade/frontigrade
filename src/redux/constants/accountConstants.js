const ADD_ACCOUNT = 'ADD_ACCOUNT';
const DELETE_ACCOUNT = 'DELETE_ACCOUNT';
const DELETE_ACCOUNTS = 'DELETES_ACCOUNTS';
const GET_ACCOUNT = 'GET_ACCOUNT';
const GET_ACCOUNTS = 'GET_ACCOUNTS';
const ADD_ACCOUNT_WIZARD = 'ADD_ACCOUNT_WIZARD';
const ADD_ACCOUNT_WIZARD_STEP_POLICY = 'ADD_ACCOUNT_WIZARD_STEP_POLICY';
const INVALID_ACCOUNT_WIZARD_STEP_POLICY = 'INVALID_ACCOUNT_WIZARD_STEP_POLICY';
const ADD_ACCOUNT_WIZARD_STEP_ROLE = 'ADD_ACCOUNT_WIZARD_STEP_ROLE';
const INVALID_ACCOUNT_WIZARD_STEP_ROLE = 'INVALID_ACCOUNT_WIZARD_STEP_ROLE';
const ADD_ACCOUNT_WIZARD_STEP_ARN = 'ADD_ACCOUNT_WIZARD_STEP_ARN';
const INVALID_ACCOUNT_WIZARD_STEP_ARN = 'INVALID_ACCOUNT_WIZARD_STEP_ARN';
const ADD_ACCOUNT_SHOW = 'ADD_ACCOUNT_SHOW';
const EDIT_ACCOUNT_SHOW = 'EDIT_ACCOUNT_SHOW';
const UPDATE_ACCOUNT_HIDE = 'UPDATE_ACCOUNT_HIDE';

export {
  ADD_ACCOUNT,
  DELETE_ACCOUNT,
  DELETE_ACCOUNTS,
  GET_ACCOUNT,
  GET_ACCOUNTS,
  ADD_ACCOUNT_WIZARD,
  ADD_ACCOUNT_WIZARD_STEP_POLICY,
  INVALID_ACCOUNT_WIZARD_STEP_POLICY,
  ADD_ACCOUNT_WIZARD_STEP_ROLE,
  INVALID_ACCOUNT_WIZARD_STEP_ROLE,
  ADD_ACCOUNT_WIZARD_STEP_ARN,
  INVALID_ACCOUNT_WIZARD_STEP_ARN,
  ADD_ACCOUNT_SHOW,
  EDIT_ACCOUNT_SHOW,
  UPDATE_ACCOUNT_HIDE
};
