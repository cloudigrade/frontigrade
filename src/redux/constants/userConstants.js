const USER_CREATE = 'USER_CREATE';
const USER_DELETE = 'USER_DELETE';
const USER_INFO = 'USER_INFO';
const USER_LOGIN = 'USER_LOGIN';
const USER_LOGOUT = 'USER_LOGOUT';

export { USER_CREATE, USER_DELETE, USER_INFO, USER_LOGIN, USER_LOGOUT };
