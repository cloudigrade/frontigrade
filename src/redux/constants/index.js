import * as toastNotificationTypes from './toastNotificationConstants';
import * as userTypes from './userConstants';

const reduxTypes = {
  toastNotifications: toastNotificationTypes,
  user: userTypes
};

export { reduxTypes as default, reduxTypes, toastNotificationTypes, userTypes };
