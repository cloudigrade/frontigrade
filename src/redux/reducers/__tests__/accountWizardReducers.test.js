import { accountWizardReducers } from '../';
import { accountTypes as types } from '../../constants';

describe('AccountWizardReducers', () => {
  it('should return the initial state', () => {
    expect(accountWizardReducers.initialState).toBeDefined();
  });

  it('should handle all defined types', () => {
    Object.keys(types).forEach(value => {
      const dispatched = {
        type: value,
        account: {
          test: 'success'
        }
      };

      const resultState = accountWizardReducers(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot('defined types');
    });
  });
});
