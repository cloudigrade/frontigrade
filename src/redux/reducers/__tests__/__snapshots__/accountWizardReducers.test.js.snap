// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined types 1`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "ADD_ACCOUNT",
}
`;

exports[`defined types 2`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "DELETE_ACCOUNT",
}
`;

exports[`defined types 3`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "DELETE_ACCOUNTS",
}
`;

exports[`defined types 4`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "GET_ACCOUNT",
}
`;

exports[`defined types 5`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "GET_ACCOUNTS",
}
`;

exports[`defined types 6`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "ADD_ACCOUNT_WIZARD",
}
`;

exports[`defined types 7`] = `
Object {
  "result": Object {
    "account": Object {
      "test": "success",
    },
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": true,
    "stepRoleValid": true,
  },
  "type": "ADD_ACCOUNT_WIZARD_STEP_POLICY",
}
`;

exports[`defined types 8`] = `
Object {
  "result": Object {
    "account": Object {
      "test": "success",
    },
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "INVALID_ACCOUNT_WIZARD_STEP_POLICY",
}
`;

exports[`defined types 9`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "ADD_ACCOUNT_WIZARD_STEP_ROLE",
}
`;

exports[`defined types 10`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "INVALID_ACCOUNT_WIZARD_STEP_ROLE",
}
`;

exports[`defined types 11`] = `
Object {
  "result": Object {
    "account": Object {
      "test": "success",
    },
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": true,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "ADD_ACCOUNT_WIZARD_STEP_ARN",
}
`;

exports[`defined types 12`] = `
Object {
  "result": Object {
    "account": Object {
      "test": "success",
    },
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "INVALID_ACCOUNT_WIZARD_STEP_ARN",
}
`;

exports[`defined types 13`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": true,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": true,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "ADD_ACCOUNT_SHOW",
}
`;

exports[`defined types 14`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "EDIT_ACCOUNT_SHOW",
}
`;

exports[`defined types 15`] = `
Object {
  "result": Object {
    "account": Object {},
    "add": false,
    "configuration": Object {},
    "edit": false,
    "error": false,
    "errorMessage": null,
    "fulfilled": false,
    "pending": false,
    "show": false,
    "stepArnErrorMessage": null,
    "stepArnValid": false,
    "stepPolicyErrorMessage": null,
    "stepPolicyValid": false,
    "stepRoleValid": true,
  },
  "type": "UPDATE_ACCOUNT_HIDE",
}
`;
